test/active_model_serializers/model_test.rb

```diff
@@ -18,5 +27,59 @@ def test_initialization_with_string_keys
 
       assert_equal model_instance.read_attribute_for_serialization(:key), value
     end
+
+    def test_attributes_can_be_read_for_serialization
+      klass = subclass_without_caching_support do
```

Comment:
But, since `id` no no longer has special behavior, subclassing without caching methods isn't necessary here and can be moved to another PR, if we like